src/modules/hip/kernel/erase.hpp

```diff
@@ -0,0 +1,386 @@
+    uint dstIdx = (id_z * dstStridesNCH.x) + (id_y * dstStridesNCH.z) + id_x;
+
+    d_float24 dst_f24;
+    rpp_hip_load24_pkd3_and_unpack_to_float24_pln3(srcPtr + srcIdx, &dst_f24);
+    rpp_hip_pack_float24_pln3_and_store24_pln3(dstPtr + dstIdx, dstStridesNCH.y, &dst_f24);
+}
+
+template <typename T>
+__global__ void convert_pln3_pkd3_hip_tensor(T *srcPtr,
+                                             uint3 srcStridesNCH,
+                                             T *dstPtr,
+                                             uint2 dstStridesNH,
+                                             RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNCH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNCH.z) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    d_float24 dst_f24;
+    rpp_hip_load24_pln3_and_unpack_to_float24_pkd3(srcPtr + srcIdx, srcStridesNCH.y, &dst_f24);
+    rpp_hip_pack_float24_pkd3_and_store24_pkd3(dstPtr + dstIdx, &dst_f24);
+}
+
+// -------------------- Set 1 - Erase main kernels --------------------
+// Rpp8u datatype
+__global__ void erase_pkd_hip_tensor(Rpp8u *dstPtr,
+                                     uint2 dstStridesNH,
+                                     RpptRoiLtrb *anchorBoxInfoTensor,
+                                     Rpp8u *colorsTensor,
+                                     Rpp32u *numBoxesTensor,
+                                     RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth))
+        return;
+
+    Rpp32u numBoxes = numBoxesTensor[id_z];
+    uchar3 pixel;
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x * 3;
+
+    // check if the co-ordinates is within any user defined box
+    for (int i = 0; i < numBoxes; i++)
+    {
+        int temp = (id_z * numBoxes) + i;
+        if (id_x >= anchorBoxInfoTensor[temp].lt.x && id_x <= anchorBoxInfoTensor[temp].rb.x && id_y >= anchorBoxInfoTensor[temp].lt.y && id_y <= anchorBoxInfoTensor[temp].rb.y)
+        {
+            temp *= 3;
+            pixel.x = colorsTensor[temp];
+            pixel.y = colorsTensor[temp + 1];
+            pixel.z = colorsTensor[temp + 2];
```

Comment:
If colorsTensor is casted in the rppt API file itelf, to either uchar3/d_schar3_s /d_half3_s /float3, you could receive that as a template U, and all these could possibly be a single templated kernel.

Reply:
This change can't be possible in rppt API file as colors should be handled differently for single and 3 channel.

Reply:
Used reinterpret_cast for pkd3 cases alone in kernel executors and it works fine.